### ISSUE TITLE
Certificate validation error using GCS

### ISSUE THREAD

**Issue by CyberHippo** [drakkan/sftpgo#430]
Hi,

I experience a certificate issue using sFTP with GCS.
I am running `sftpgo` on kubernetes 1.18 (GKE) with the image `drakkan/sftpgo:v2.0.4-alpine` and postgreSQL as _data-provider_.  The service starts up nicely.
I created a user and uploaded a JSON file for the _Credentials file_ and I specified the bucket name (I omited the storage class) using the web interface. Then I initiated the connection to the sftpgo service successfully:
```json
# sftpgo logs
{"level":"info","time":"2021-05-17T15:43:13.533","sender":"SSH","connection_id":"50874aa32785f312c4e04XXX","message":"User id: 1, logged in with: \"password\", username: \"test\", home_dir: \"/xxx\" remote addr: \"xx.xx.xx.xx\""}
{"level":"debug","time":"2021-05-17T15:43:13.538","sender":"dataprovider_postgresql","message":"last login updated for user \"test\""}
{"level":"debug","time":"2021-05-17T15:43:13.538","sender":"common","connection_id":"50874aa32785f312c4e04XXX","message":"ssh connection added, num open connections: 1"}
{"level":"debug","time":"2021-05-17T15:43:13.567","sender":"SFTP","connection_id":"SFTP_50874aa32785f312c4e04XXX_1","message":"connection added, num open connections: 1"}

# Client logs
Connected to xx.xx.xx.xx.
```

Then I tried to list the content of the bucket and I got the following errors:
```sh
# Client logs
sftp> ls
remote readdir("/"): Failure

# sftpgo logs
{"level":"warn","time":"2021-05-17T15:44:51.211","sender":"SFTP","connection_id":"SFTP_50874aa32785f312c4e04XXX_1","message":"error listing directory: Get \"https://storage.googleapis.com/storage/v1/b/<bucket-name>/o?alt=json&delimiter=%2F&endOffset=&fields=nextPageToken%2Cprefixes%2Citems%28contentType%2Cname%2Csize%2CtimeDeleted%2Cupdated%29&pageToken=&prefix=&prettyPrint=false&projection=full&startOffset=&versions=false\": oauth2: cannot fetch token: Post \"https://oauth2.googleapis.com/token\": x509: certificate signed by unknown authority"}
```

Do you have any idea why the `sftpgo` service cannot verify the certificate ? 
Thank you for your help.

---

PS: Thank you for the amazing work on this project. :rocket:

**Comment by drakkan** [comment]
Hi,

I just tested this way:

```
docker run --name some-sftpgo -p 8080:8080 -p 2022:2022 "drakkan/sftpgo:v2.0.4-alpine"
```

I created a GCS user the same way you described and was able to list the contents of the bucket.

I think this error occurs if the `ca-certificates` package is not installed inside the docker image but the official docker image [installs it](https://github.com/drakkan/sftpgo/blob/main/Dockerfile.alpine#L34). 

Can you please confirm that your docker image has CA certificates installed? Thank you

**Comment by CyberHippo** [comment]
Hi @drakkan,

Thank you for your prompt reply and kind help.

I tested it locally the same way you mentionned in your reply and it worked nicely.

I also checked that the `ca-certificates` package was not missing from image and it is present.
```sh
$ apk list
[...]
ca-certificates-bundle-20191127-r4 x86_64 {ca-certificates} (MPL-2.0 GPL-2.0-or-later) [installed]
```

I'll close the issue for now as the issue does not seem to be related to `sftpgo` and will continue to investigate further on my GKE cluster. Thanks again for your time.
